play-json/shared/src/main/scala/play/api/libs/json/Writes.scala

```diff
@@ -265,15 +269,48 @@ trait DefaultWrites extends LowPriorityWrites {
   /**
    * Serializer for Map[String,V] types.
    */
+  @deprecated("Use `genericMapWrites`", "2.7.0")
   implicit def mapWrites[V: Writes]: OWrites[MapWrites.Map[String, V]] = MapWrites.mapWrites
 
   /**
-   * Serializer for JsValues.
+   * Serializer for Map[String,V] types.
    */
-  implicit object JsValueWrites extends Writes[JsValue] {
+  implicit def genericMapWrites[V, M[A, B] <: MapWrites.Map[A, B]](implicit w: Writes[V]): OWrites[M[String, V]] = OWrites[M[String, V]] { ts =>
+    JsObject(ts.mapValues(w.writes(_)).toSeq)
+  }
+
+  @deprecated("Use `jsValueWrites`", "2.7.0")
+  object JsValueWrites extends Writes[JsValue] {
     def writes(o: JsValue) = o
   }
 
+  /**
+   * Serializer for JsValues.
+   */
+  implicit def jsValueWrites[T <: JsValue]: Writes[T] = Writes[T] { js => js }
+
+  /**
+   * Serializer for JsNull.
+   *
+   * {{{
+   * Json.obj("foo" -> None)
+   * // equivalent to Json.obj("foo" -> JsNull)
+   * }}}
+   */
+  implicit val NoneWrites: Writes[None.type] =
+    Writes[None.type] { _ => JsNull }
+
+  /**
+   * If `Some` is directly used (not as `Option`).
+   *
+   * {{{
+   * Json.obj("foo" -> Some(writeableValue))
+   * // equivalent to Json.obj("foo" -> writeableValue)
+   * }}}
+   */
+  implicit def someWrites[T](implicit w: Writes[T]): Writes[Some[T]] =
+    Writes[Some[T]] { some => w.writes(some.get) }
```

Comment:
Should we handle the case where `some.get` returns `null`, or should this be delegated to the implicit `Writes[T]`?

Reply:
Nothing specific there, on this PR or one master:

```scala
scala> Json.toJson(Some(null: String))
res0: play.api.libs.json.JsValue = null

scala> Json.toJson(Some(null: String)).getClass
res1: Class[_ <: play.api.libs.json.JsValue] = class play.api.libs.json.JsString
```